packages/contracts/src/File.sol

```diff
@@ -12,26 +12,28 @@ struct File {
 }
 
 function read(File memory file) view returns (string memory contents) {
-    address[] memory pointers = new address[](file.contents.length);
-    for (uint256 i = 0; i < file.contents.length; i++) {
-        pointers[i] = file.contents[i].pointer;
-    }
+    Content[] memory chunks = file.contents;
```

Comment:
There must be a way to access this part of the `file` object in assembly without having to create a reference to it here.